### PR TITLE
Improved test coverage.

### PR DESCRIPTION
**Pull request by QKaiser**
Improved our test coverage by implementing unit tests for FAT handler and recently introduced entropy visualisation functions.

### CODE REVIEW

**Review by kukovecz — COMMENTED** [review]
Added a few remarks. Otherwise, really cool intuitive, I always like adding tests, keep up the good work :smile:

File: tests/test_processing.py [hunk 1]
```diff
@@ -90,3 +97,53 @@ def test_calculate_unknown_chunks(
...
+
+@pytest.mark.parametrize(
+    "percentages",
+    [
+        pytest.param([], id="empty-array"),
+    ],
+)
+def test_draw_entropy_plot_error(percentages: List[float]):
+    with pytest.raises(TypeError):
+        draw_entropy_plot(percentages)
```
[the remark this answers is not in the document]
Comment (QKaiser) [reply tests/test_processing.py]: Fixed.

File: tests/test_processing.py [hunk 2]
```diff
@@ -90,3 +97,53 @@ def test_calculate_unknown_chunks(
...
+
+
+@pytest.mark.parametrize(
+    "path, draw_plot, expected",
+    [
+        pytest.param(Path("/proc/self/exe"), True, None, id="draw-plot"),
+        pytest.param(Path("/proc/self/exe"), False, None, id="no-plot"),
+    ],
+)
+def test_calculate_entropy(path: Path, draw_plot: bool, expected):
```
[the remark this answers is not in the document]
Comment (QKaiser) [reply tests/test_processing.py]: Fixed.

File: tests/test_processing.py [hunk 3]
```diff
@@ -90,3 +97,53 @@ def test_calculate_unknown_chunks(
...
+
+@pytest.mark.parametrize(
+    "percentages, expected",
+    [
+        pytest.param([0.0] * 100, None, id="zero-array"),
+        pytest.param([99.99] * 100, None, id="99-array"),
+        pytest.param([100.0] * 100, None, id="100-array"),
+    ],
+)
+def test_draw_entropy_plot(percentages: List[float], expected):
```
[the remark this answers is not in the document]
Comment (QKaiser) [reply tests/test_processing.py]: Fixed.

File: tests/handlers/filesystem/test_fat.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    io.BytesIO(VALID_FAT16_HEADER_CONTENT)
+)
+VALID_FAT32_HEADER = handler.cparser_le.fat32_bootsec_t(
+    io.BytesIO(VALID_FAT32_HEADER_CONTENT)
+)
+
+# we have to create unique objects but shallow or deep copy triggers a
+# recursion error on cstruct.Instance objects so we just parse the
+# valid header again then modify it to test our specific checks.
+FAT12_HEADER_INVALID_OEM = handler.cparser_le.fat12_16_bootsec_t(
```
[the remark this answers is not in the document]
Comment (QKaiser) [reply tests/handlers/filesystem/test_fat.py]: Definitely more legible this way.

**Review by kukovecz — APPROVED** [review]

**Pull request merged**